clicommand/agent_start.go

```diff
@@ -552,7 +552,11 @@ var AgentStartCommand = cli.Command{
 
 		// Set a useful default for the bootstrap script
 		if cfg.BootstrapScript == "" {
-			cfg.BootstrapScript = fmt.Sprintf("%s bootstrap", shellwords.Quote(os.Args[0]))
+			exePath, err := os.Executable()
+			if err != nil {
+				l.Fatal("Unable to find executable path for bootstrap")
```

Comment:
I'm not  sure what kind of error might happen :man_shrugging:

Reply:
Yeah, it would probably be unusual / edge-case / exotic errors, given it's implemented differently for plan9, freebsd, solaris, darwin, linux/procfs, and windows. Some of them mention potential buffer size problems, I guess the procfs one depends on procfs not being broken and/or having the right permissions to read `/proc/self/exe`, etc.

Reply:
e.g. running buildkite-agent in a poorly-configured chroot on my Kindle, I encountered all sorts of errors that wouldn't normally happen, e.g. when `/dev/random` doesn't exist or isn't accessible, and I may have hit some procfs issues too.